### DISCUSSION TITLE
invalid go:embed: build system did not supply embed configuration

### DISCUSSION THREAD

**Discussion by Jeeyuen-Rick** [golangci/golangci-lint#4984] [Q&A] [answered]
I upgraded Go to 1.23.0 and golangci-lint 1.60.3, I get the following error message with existing project:

```
typechecking error: invalid go:embed: build system did not supply embed configuration\<path>/myfile.go:18:3: invalid go:embed: build system did not supply embed configuration"
```

I used //go:embed directive in my programs. Is there a way to fix this issue or workaround this issue? I don't have this linting error when my project is using Go 1.22.4 and golangci-lint 1.56.2.

**Comment by ldez** [comment, marked as answer]
Hello, this error is not new but I guess you have tried to skip/ignore it with an exclude rule.

As explained inside the [documentation](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors), linters cannot perform the analysis when there are `typecheck` errors. This is not related to golangci-lint itself but to how the analysis system works inside Go tooling.

`typecheck` errors are reported in the same style as linter reports/issues but are completely different because they are related to compilation problems (`typecheck` is not a linter).

The [change](https://golangci-lint.run/product/changelog/#v1601) inside v1.60.1 was to avoid side effects: most linters are not able to perform the analysis when there are `typecheck` errors, but some users thought it was working to just skip `typecheck` errors.

To be more explicit, before this change, most linters simply did not produce any reports because of the presence of `typecheck` errors.

So `typecheck` errors cannot be ignored/skipped.

The only way to "fix" is to provide the file required by the embed directive.

**Comment by Jeeyuen-Rick** [comment]
Thank you for the quick response and https://golangci-lint.run/product/changelog/#v1601 is helpful.

**Comment by cbandy** [comment]
I encountered this and found that it happens when the file I wanted to embed does not exist AND the `golangci-lint` cache is cleared.

1. When the embed file exists, `golangci-lint run` succeeds.
2. If I delete the embed file, `golangci-lint run` still succeeds.
3. If I `golangci-lint cache clean`, then `golangci-lint run` fails until I create the embed file.
